src/dragonfly/dragonfly/actions/SketchAction.py

```diff
@@ -324,6 +324,12 @@ def broadcast_target(self, partner_position, self_position):
             self.broadcast_end()
             self.end_sketch()
 
+        if self.target_position_vector.movement == PositionVector.TURN and self.target_position_vector.a * self.target_position_vector.p > 4 * math.pi:
+            # We are in a death spiral
+            self.logger.info("Sketch ended")
```

Comment:
We should probably update this info to indicate the failure saying something like adjust threshold or lambda

Reply:
Good idea...